connect/api/src/main/java/org/apache/kafka/connect/data/Values.java

```diff
@@ -811,14 +840,19 @@ protected static SchemaAndValue parse(Parser parser, boolean embedded) throws No
                 if (COMMA_DELIMITER.equals(parser.previous())) {
                     throw new DataException("Map is missing element after ',': " + parser.original());
                 }
-                throw new DataException("Map is missing terminating ']': " + parser.original());
+                throw new DataException("Map is missing terminating '}': " + parser.original());
             }
         } catch (DataException e) {
-            LOG.debug("Unable to parse the value as a map; reverting to string", e);
+            LOG.debug("Unable to parse the value as a map or an array; reverting to string", e);
             parser.rewindTo(startPosition);
         }
-        String token = parser.next().trim();
-        assert !token.isEmpty(); // original can be empty string but is handled right away; no way for token to be empty here
```

Comment:
This seems like an assumption was made with conviction, did you see this during testing?
Did the other code change in a way that empty strings are possible here now?

Reply:
I did change the code in a way that empty strings are possible here, and yeah, that was discovered through testing.